.github/workflows/go.yml

```diff
@@ -0,0 +1,30 @@
+name: Build Lint and Test
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+  workflow_dispatch:
+
+jobs:
+  build-lint-test:
+    name: Build Lint and Test
+    runs-on: ubuntu-24.04
+    steps:
+      - uses: actions/checkout@v4
+      - name: Setup Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: '1.17'
+      - name: Build
+        run: make build
+      - name: Linting
+        run: make check
```

Comment:
@CodeLieutenant check doesn't exist, and no this is failing across the board, can you fix this please ?